src/components/ButtonComponent/ButtonComponent.tsx

```diff
@@ -0,0 +1,17 @@
+// src/components/Button/Button.tsx
+import styles from './ButtonComponent.module.css';
+// import React from 'react';
+import { MoonLoader } from 'react-spinners';
+import type { ButtonProps } from './ButtonComponent.definitions';
+
+const Button = ({ variant, color, onClick, text, loading }: ButtonProps) => {
+    const buttonClassName = `${styles.button} ${styles[variant]} ${styles[color]}`;
+
+    return (
+        <button className={buttonClassName} onClick={onClick} disabled={loading}>
+            {loading ? <MoonLoader size={30} color="#FFFFFF" /> : text}
```

Comment:
It appears the `MoonLoader` does not show properly. The button should look like this:

#### Loading
<img width="170" alt="Screen Shot 2023-11-19 at 9 10 07 PM" src="https://github.com/ChangePlusPlusVandy/miracleflights-frontend/assets/19592236/58f73d27-10c1-43c5-b8b0-8aebb3a8f987">

#### Icon
<img width="170" alt="Screen Shot 2023-11-19 at 9 08 45 PM" src="https://github.com/ChangePlusPlusVandy/miracleflights-frontend/assets/19592236/abe7ffa0-a825-4244-ba72-ea0c15ab446f">

#### No Icon
<img width="146" alt="Screen Shot 2023-11-19 at 9 11 14 PM" src="https://github.com/ChangePlusPlusVandy/miracleflights-frontend/assets/19592236/d991c52f-01ef-499b-a538-0969eb3e49d9">